trpcultivate_phenotypes/src/ListBuilder/PhenodataBackupListBuilder.php

```diff
@@ -0,0 +1,348 @@
+    $build['table'] = parent::render();
+
+    return $build;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function load() {
+
+    $query = $this->getEntityListQuery();
+
+    if (!isset($this->entity_field_header['user_id'])) {
+      $query
+        ->condition('user_id', $this->user->id());
+    }
+
+    $filter_project_id = \Drupal::request()
+      ->get('project_id', 0);
+
+    // Any attempt to mangle with the query string will just default to
+    // load all projects.
+    if (!preg_match('/^\d+$/', (string) $filter_project_id)) {
+      $filter_project_id = 0;
+    }
+
+    if ($filter_project_id > 0) {
+      $query
+        ->condition('project_id', $filter_project_id, '=');
+    }
+
+    $query->sort('backup_date', 'DESC');
+    $entity_ids = $query->execute();
+
+    return $this->storage
+      ->loadMultiple($entity_ids);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getFormId() {
+
+    return 'phenodata_backup_display_form';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    $form = [];
+    // Panels tend to stick, apply some margin.
+    $form['#attributes']['style'] = 'margin-bottom: 20px';
+
+    // Populate the select field with project names.
+    $project_names = [];
+
+    $query = $this->getEntityListQuery();
+
+    if (!isset($this->entity_field_header['user_id'])) {
+      $query
+        ->condition('user_id', $this->user->id());
+    }
+
+    $entity_ids = $query->execute();
+    $list = $this->storage->loadMultiple($entity_ids);
+
+    $project_names = [
+      0 => '- Any - ',
+    ];
+
+    foreach ($list as $entity) {
+      $project_id = (int) $entity->get('project_id');
+      $project_names[$project_id] = ChadoProjectAutocompleteController::getProjectName($project_id);
+    }
+
+    $filter_project_id = \Drupal::request()
+      ->get('project_id', 0);
+
+    $form['project_id'] = [
+      '#type' => 'select',
+      '#title' => $this->entity_field_header['project_id'],
+      '#options' => $project_names,
+      '#attributes' => [
+        'style' => 'width: 100%',
+      ],
+      '#default_value' => $filter_project_id,
+    ];
+
+    $form['actions'] = [
+      '#tree' => FALSE,
+      '#type' => 'actions',
+    ];
+
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => 'Filter',
+      '#button_type' => 'default',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+
+    $project_id = (int) $form_state->getValue('project_id');
+
+    if ($project_id > 0) {
+      $project_name = ChadoProjectAutocompleteController::getProjectName($project_id);
+
+      if (empty($project_name)) {
+        $form_state->setErrorByName('project_id', $this->t(
+          'The @project is not recognized. Start by typing slowly, select the correct one from the dropdown and try saving again.',
```

Comment:
While mannually testing, I've realized that I can't trigger this message. I should have made my code suggestion in `PhenodataBackupForm:validateForm()` instead of here.

I think this message should be updated to be accurate in the event that it does get triggered. Perhaps something like:
```suggestion
          'The @project is not recognized or you do not have permission to see backups for it.',
```